smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/generators/FromNodeMapperVisitor.java

```diff
@@ -112,6 +112,7 @@ public Void floatShape(FloatShape shape) {
 
     @Override
     public Void documentShape(DocumentShape shape) {
+        writer.write("$L.expectObjectNode()", varName);
```

Comment:
A document can also be a string or any other value. Would this still work for those cases?

Reply:
Good point. The document needs to be a `Node` not `ObjectNode`. Then this can just be changed to transparently pass the node here.